plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -269,55 +296,79 @@ function OPDSBrowser:fetchFeed(item_url, username, password, method)
         password = password,
     }
     logger.info("Request:", request)
+    -- Fire off the request and wait to see what we get back.
     local code, headers = socket.skip(1, http.request(request))
     socketutil:reset_timeout()
-    -- raise error message when network is unavailable
+    -- Check the response and raise error message when network is unavailable.
     if headers == nil then
         error(code)
     end
+    -- Below are numerous if cases to handle different response codes.
     if code == 200 then
+        -- 200 means the request succeeded.
+        -- If the method sent was HEAD, then we're probably checking for
+        -- an update and therefore only interested in the last-modified
+        -- time of the resource (who needs a body when you have a head?).
         if method == "HEAD" then
             if headers["last-modified"] then
                 return headers["last-modified"]
             else
                 return
             end
         end
+        -- If the method sent was not HEAD, then we are interested in
+        -- the payload of the request. We'll add that to a table below
+        -- and return that as the result of this function.
         local xml = table.concat(sink)
+        -- Obviously, check to see if the payload exists.
         if xml ~= "" then
             return xml
         end
     elseif method == "HEAD" then
         -- Don't show error messages when we check headers only.
         return
-    elseif code == 301 then
-        UIManager:show(InfoMessage:new{
-            text = T(_("The catalog has been permanently moved. Please update catalog URL to '%1'."), BD.url(headers['Location'])),
+    elseif code == 301 then -- Page has permanently moved
+        UIManager:show(
+            InfoMessage:new{
```

Comment:
These are also kind of the same thing. But you say you're in favor @NiLuJe? I'm not necessarily opposed but it's certainly a different style.

Reply:
It takes a single argument, clearer to have it on the same line.